da/lazyledger/lazyledger.go

```diff
@@ -80,14 +88,57 @@ func (ll *LazyLedger) SubmitBlock(block *types.Block) da.ResultSubmitBlock {
 		return da.ResultSubmitBlock{Code: da.StatusError, Message: err.Error()}
 	}
 
-	err = ll.callRPC(msg)
+	ctx, cancel := context.WithTimeout(context.TODO(), ll.config.Timeout)
+	defer cancel()
+
+	err = ll.callRPC(ctx, msg)
 	if err != nil {
 		return da.ResultSubmitBlock{Code: da.StatusError, Message: err.Error()}
 	}
 
 	return da.ResultSubmitBlock{Code: da.StatusSuccess}
 }
 
+func (ll *LazyLedger) callRPC(ctx context.Context, msg *apptypes.MsgWirePayForMessage) error {
+	info, err := ll.keyring.Key(ll.config.KeyringAccName)
+	if err != nil {
+		return err
+	}
+
+	b := appclient.NewBuilder(info.GetAddress(), ll.config.ChainID)
+
+	err = b.UpdateAccountNumber(ctx, ll.rpcClient)
+	if err != nil {
+		return err
+	}
+
+	coin := sdk.Coin{
+		Denom:  "token",
+		Amount: sdk.NewInt(10),
+	}
+	b.SetFeeAmount(sdk.NewCoins(coin))
+
+	b.SetGasLimit(ll.config.GasLimit)
+
+	signedTx, err := b.BuildSignedTx(msg, ll.keyring)
+	if err != nil {
+		return err
+	}
+
+	rawTx, err := b.EncodeTx(signedTx)
+	if err != nil {
+		return err
+	}
+
+	// wait for the transaction to be confirmed in a block
+	_, err = appclient.BroadcastTx(ctx, ll.rpcClient, tx.BroadcastMode_BROADCAST_MODE_BLOCK, rawTx)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we want to wait until the tx is included in the block, but I'm not sure. Is this okay? 

Our other options are here https://github.com/lazyledger/cosmos-sdk/blob/3addd7f65d1c35f76716d67641ee629b01f5b9c7/types/tx/service.pb.go#L37